### PR TITLE
feat: add unit tests config and examples

### PR DESCRIPTION
**Pull request by Helias**
I would like to apply the same directory structure to Spotted-DMI-Bot and keep only the unit-test in the pipeline, so the e2e tests will be not part of the pipeline, unfortunately, they are really slow (locally and through GitHub action), moreover now they need some updates after the multi-lang feature.

### CODE REVIEW

File: .github/workflows/telegram-bot.yml [hunk 1]
```diff
@@ -37,13 +42,14 @@ jobs:
...
-          pytest
-
+      - name: Unit test
+        run: pytest tests/unit/
+      # - name: End-to-end test
+      #   run: pytest tests/e2e/
 
   doc:
     needs: test
+    if: github.ref == 'refs/heads/master'
```
Comment (dcariotti) [inline .github/workflows/telegram-bot.yml:52]: Why don't you use two unique workflows? One for any push and one only for push to `master` branch.
```
.github/workfows
├── tests.yml
└── docs.yml
```

File: tests/unit/test_professor.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from module.data.professor import Professor
+
+def test_professor_table() -> None:
+    professor = Professor()
+    assert professor.table == "professors"
```
[the remark this answers is not in the document]
Comment (dcariotti) [reply tests/unit/test_professor.py:5]: I agree with that. Let's consider this file as an example for the future tests :+1:

File: .github/workflows/telegram-bot.yml (hunk 1 above)
Comment (Helias) [reply .github/workflows/telegram-bot.yml:52]: `needs: test` will work anyway? or should I remove it?

Comment (dcariotti) [reply .github/workflows/telegram-bot.yml:52]: If a PR is merged into `master`, you know the `tests` have already passed :smiley:

**Review by TendTo — APPROVED** [review]

File: .github/workflows/telegram-bot.yml (hunk 1 above)
Comment (Helias) [reply .github/workflows/telegram-bot.yml]: now, yes 😁

**Review by dcariotti — APPROVED** [review]
LGTM

**Pull request merged**